test/functional/core/exit_spec.lua

```diff
@@ -43,4 +43,17 @@ describe('v:exiting', function()
     run(on_request, nil, on_setup)
   end)
 
+  it('is specified exit code after :cquit', function()
+    local function on_setup()
+      command('autocmd VimLeavePre * call rpcrequest('..cid..', "")')
+      command('autocmd VimLeave    * call rpcrequest('..cid..', "")')
+      command('cquit 123')
+    end
+    local function on_request()
+      eq(123, eval('v:exiting'))
+      return ''
+    end
+    run(on_request, nil, on_setup)
+  end)
```

Comment:
Also it is good idea to test whether executing `if 0|cquit 5|endif` throws some syntax errors (better use `redir_exec`, not `exc_exec`). With some combination of flags you need to bother populating `eap->nextcmd`, though I do not remember with which exactly (most likely not `WORD1` though and definitely not with COUNT+NOTADR, but without WORD1).

Reply:
And check `:0cquit` and `:cquit 0`. It is logical that second will yield zero exit code and illogical, but consistent that first will yield exit code 1 (check `:0cnext` vs `:cnext 0`).

Reply:
Ah, no, add ZEROR flag to the list and `:0cquit` should be the same as `:cquit 0`. It is just `:cnext` thinks that is incorrect.

Reply:
And COUNT goes with RANGE for desired behaviour. Basically what is expected is

```patch
diff --git a/src/nvim/ex_cmds.lua b/src/nvim/ex_cmds.lua
index 5a578cd08..6a1e6ceb3 100644
--- a/src/nvim/ex_cmds.lua
+++ b/src/nvim/ex_cmds.lua
@@ -614,7 +614,7 @@ return {
   },
   {
     command='cquit',
-    flags=bit.bor(TRLBAR, BANG),
+    flags=bit.bor(RANGE, NOTADR, COUNT, ZEROR, TRLBAR, BANG),
     addr_type=ADDR_LINES,
     func='ex_cquit',
   },
```